vim/syntax/vimtypingfilter.vim

```diff
@@ -0,0 +1,20 @@
+"" include underlying syntax file as denoted by syntaxbase variable
+"" keep value of b:current_syntax variable
+execute "runtime! " . fnameescape("syntax/" . syntaxbase . ".vim")
+
+"" add a new syntax group called vimTypingTeXTags and allow it to be matched
+"" within other groups as well as on top
+
+"" TODO this is a rather broad matching guess
+"" some languages need to be setup more specific eg. HTML
+"" single them out by questing explicitly syntaxbase string defining
+"" vimTypeingTeXTags syntax group specifically for language instead of using
+"" below generic pattern
+"" TODO the below generic group-name pattern covering a big set of languages
+"" rather well. It is rather non specific global
+"" make more specific if necessary
+syntax region vimTypingTeXTags start="/BTEX" end="/ETEX" transparent containedin=\w*\([Cc]omment\|[Tt]odo\|[Ss]tatement\|[Mm]acro\|[Dd]efine\|[Pp]re[Cc]ondit\|[Tt]ypedef\|[Dd]ebug\|[Bb]lock\|[Rr]egion\|[Hh]eredoc\|[Ee]nvironment\|[Ee]xpression\|[Cc]ontext\|[Dd]ocument\|[Zz]onei\|[Bb]ody\|[Ss]kip\)\w* contains=NONE
```

Comment:
For the simple tests, I can simplify this to `containedin=ALL`. Does that also works in your bigger examples?

Reply:
I have thought about using contained in ALL. The idea why i refrained from it was that there are parts of the source code where cluttering with `/BTEX` `/ETEX` doesent make sense to me:
   Strings: should not contain any stuff which is typeset specially by ConTeXt deviatint its appearance from what is possible and expected by the programming language. Further using `/BTEX` `/ETEX` therein for marking lines with `\someline`, `\startline` and `\stopline` reduces the readablility of these strings and other elements.
   Other: where readability of the source would become quite difficult

Therefore my next idea was to use `ALLEXEMPT` and than just specify the few regions which should be excempt.

I tossed this too as soon as i figured from looking at vim syntaxfiles how diverse between different languages string constants and alike are indicated and how diverse the names of the regions and corresponding syntax elements are. The pattern would have been a lot longer just for a small ammount of standard languages. 

Therefore I finally resettled to an explicit white listing as this seemed too me the second best option if `ALL` is too permissive.

Having sayed this `ALL` is OK if documentation clearly states that eventhough possible sections bracketed by `/BTEX`, `/ETEX` should be avoided within strings and other alike constants and statements where typesetting by ConTeXt would cause compiler or interpreter errors when readers copy and paste the code to a source file or interactive environment to try it.

It is your decision which rout to go.

Reply:
I think that `ALL`  may be simpler. I really want to avoid any dependencies on the sytanxfile use in the module.